aten/src/ATen/native/cuda/ReflectionPadding1d.cu

```diff
@@ -0,0 +1,215 @@
+#include <ATen/ATen.h>
+#include <ATen/cuda/CUDAApplyUtils.cuh>
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/NativeFunctions.h>
+#include <ATen/TensorUtils.h>
+#include <ATen/Utils.h>
+// keeping THC headers for atomicAdd
+#include <THC/THCGeneral.h>
+
+namespace at {
+namespace native {
+namespace {
+
+using at::cuda::detail::canUse32BitIndexMath;
+
+__device__
+inline void get_index_mapping(
+    int64_t input_w, int64_t output_w,
+    int64_t output_x,
+    int64_t pad_l,
+    int64_t & input_idx, int64_t & output_idx) {
+  // 3D grid of 1D blocks
+  auto input_offset =
+    (blockIdx.y + blockIdx.z * gridDim.y) * input_w;
+  auto output_offset =
+    (blockIdx.y + blockIdx.z * gridDim.y) * output_w;
+
+  auto i_start_x = ::max(0L, (long)-pad_l);
```

Comment:
does `::max(int64_t(0), -pad_l)` work?  (and for the line below)?

Reply:
Let me try.